source/unified-test-format/unified-test-format.rst

```diff
@@ -208,6 +209,11 @@ ChangeStream) and only allow access to BSON types (see:
 test runner needs to ensure driver objects in its entity map are properly
 freed/destroyed between tests.
 
+The entity map MUST be implemented in a way that allows for safe concurrent
+access, since a test may include multiple thread entities that all need to
```

Comment:
What kind of concurrent safe access is needed? Will operations be overwritting the same entity in the Map at any point? 

I think that we can expect test operations to always be writing to distinct keys and only reading from shared clients/sessions etc. Is it worth calling out the specifics around what a driver can expect a test to be doing to the map concurrently?

Reply:
Whatever an operation can do to the entity map (e.g. read from it, modify an existing entity, add a new one) can happen at the same time from an arbitrary number of contexts via runOnThread.

The text here is mostly a heads-up to implementors so that they don't accidentally introduce data races into their runners when adding support for thread entities. I don't think we need to go into too much detail about the specific types of concurrent access and can leave it to implementations to determine what level of safety they need.